src/lib/optimizer/optimizer.cpp

```diff
@@ -136,9 +77,9 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   // StoredTableNode as possible where the ChunkPruningRule can work with them.
   optimizer->add_rule(std::make_unique<ChunkPruningRule>());
 
-  // This is an optimization for the PQP sub-plan memoization which is sensitive to the a StoredTableNode's table name,
-  // set of pruned chunks and set of pruned columns. Since this rule depends on pruning information, it has to be
-  // executed after the ColumnPruningRule and ChunkPruningRule.
+  // This is an optimization for better PQP subplan deduplication in the LQPTranslator. It aligns all StoredTableNodes
+  // that refer to the same table and the same set of pruned chunk ids. It is sensitive to all pruning information
+  // and has therefore to be executed after the ColumnPruningRule and the ChunkPruningRule.
```

Comment:
Suggestion:

> The LQPTranslator deduplicates subplans to avoid performing the same computation twice (see LQPTranslator::translate_node). The StoredTableColumnAlignmentRule supports this effort by aligning the list of pruned columns across nodes that could become deduplicated. For this, the ColumnPruningRule needs to have been executed.

Reply:
Add a hint, that the ChunkPruningRule should better be executed before the STNColumnAlignmentRule?